### PR TITLE
SSL_new_session_ticket() support (1.1.1)

### PR DESCRIPTION
**Pull request by kaduk**
Backport the core `SSL_new_session_ticket()` API from OpenSSL 3.0.0, plus the recent changes from quictls/openssl-3.0.0-alpha14+quic that allow it to work with QUIC.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Review by nibanks — APPROVED** [review]

**Comment by nibanks** [comment]
@kaduk are you just waiting on additional reviews before merging this?

**Comment by kaduk** [comment]
Yes; I wasn't sure if @tmshort wanted to approve as well.

File: util/libssl.num
Comment (tmshort) [inline util/libssl.num:517]: This is ok if we assume that this function won't eventually be added to 1.1.1. Do we think it will be?

Comment (kaduk) [reply util/libssl.num:517]: The new API implements a new feature, so it would take an OTC vote to approve an exception policy in order for it to make it to 1.1.1.  I don't expect that to happen (though I can't definitively rule it out).

Comment (tmshort) [reply util/libssl.num:517]: That's fair.

**Review by tmshort — APPROVED** [review]

**Comment by nibanks** [comment]
Thanks for merging this! Just merged this into MsQuic with https://github.com/microsoft/msquic/pull/1469. All the tests are passing.

**Pull request merged**
